### ISSUE TITLE
Support docker container runtime in Kubernetes 1.24

### ISSUE THREAD

**Issue by afbjorklund** [kubernetes/minikube#13545]
This is a tracker for the required changes:

* https://github.com/kubernetes/minikube/pull/12102

* https://github.com/kubernetes/minikube/pull/13191

* https://github.com/kubernetes/minikube/pull/13227
* https://github.com/kubernetes/minikube/pull/13302

* https://github.com/kubernetes/minikube/pull/13563

`minikube start --kubernetes-version=v1.24.0-alpha.2 --container-runtime=docker`

Failing with minikube v1.25.1, due to the earlier confusion about the socket name

ISO (missing)
```
Failed to enable unit: Unit file cri-docker.socket does not exist.
```

KIC (outdated)
```
stat: cannot stat '/var/run/cri-dockerd.sock': No such file or directory
```

```console
docker@minikube:~$ systemctl cat cri-docker.socket 
# /lib/systemd/system/cri-docker.socket
[Unit]
Description=CRI Docker Socket for the API
PartOf=cri-docker.service

[Socket]
ListenStream=%t/cri-docker.sock
SocketMode=0660
SocketUser=root
SocketGroup=docker

[Install]
WantedBy=sockets.target
docker@minikube:~$ cri-dockerd --version
cri-dockerd 0.2.0 (HEAD)
```

* https://github.com/Mirantis/cri-dockerd/commit/50c048cb54e52cd9058f044671e309e9fbda82e4

```diff
@@ -3,7 +3,7 @@ Description=CRI Docker Socket for the API
 PartOf=cri-docker.service
 
 [Socket]
-ListenStream=%t/cri-docker.sock
+ListenStream=%t/cri-dockerd.sock
 SocketMode=0660
 SocketUser=root
 SocketGroup=docker
```

Still wrong in the upstream README, but there it is the unit and not the file.

* https://github.com/Mirantis/cri-dockerd/pull/33

**Comment by afbjorklund** [comment]
@sharifelgamal : added tracker issue, for the required PRs

**Comment by afbjorklund** [comment]
v1.24.0-alpha.1

```go
	// DefaultDockerCRISocket defines the default Docker CRI socket
	DefaultDockerCRISocket = "/var/run/dockershim.sock"
```

v1.24.0-alpha.2

```go
	// CRISocketDocker is the cri-dockerd CRI endpoint
	CRISocketDocker = "unix:///var/run/cri-dockerd.sock"
```

**Comment by afbjorklund** [comment]
My bad, thought the socket path fix was included in `v0.2.0` but it was in v0.2.0-7-g3eee3e4

Can go straight for v0.2.0-11-ga4d1895 and get some fixes for the README and --buildinfo ?
